src/renderer/renderer_elements.ts

```diff
@@ -550,134 +656,159 @@ export class Edge extends SDFGElement {
-            } else if (num_accesses == '-1') {
-                num_accesses = '<b>Dynamic (unbounded)</b>';
-            }
+        if (attr.wcr)
+            contents += '<br /><b>CR: ' + sdfg_property_to_string(
+                attr.wcr, dsettings
+            ) + '</b>';
 
-            contents += '<br /><font style="font-size: 14px">Volume: ' +
-                num_accesses + '</font>';
-            container.innerHTML = contents;
-        } else {  // Interstate edge
-            container.classList.add('sdfvtooltip--interstate-edge');
-            container.innerText = this.label();
-            if (!this.label())
-                container.style.display = 'none';
+        let num_accesses = null;
+        if (attr.volume)
+            num_accesses = sdfg_property_to_string(attr.volume, dsettings);
+        else
+            num_accesses = sdfg_property_to_string(
+                attr.num_accesses, dsettings
+            );
+
+        if (attr.dynamic) {
+            if (num_accesses === '0' || num_accesses === '-1')
+                num_accesses = '<b>Dynamic (unbounded)</b>';
+            else
+                num_accesses = '<b>Dynamic</b> (up to ' +
+                    num_accesses + ')';
+        } else if (num_accesses === '-1') {
+            num_accesses = '<b>Dynamic (unbounded)</b>';
         }
-    }
 
-    public set_layout(): void {
-        // NOTE: Setting this.width/height will disrupt dagre in self-edges
+        contents += '<br /><font style="font-size: 14px">Volume: ' +
+            num_accesses + '</font>';
+        container.innerHTML = contents;
     }
 
     public label(): string {
-        // Memlet
-        if (this.data.attributes.subset !== undefined)
-            return '';
-        return super.label();
+        return '';
     }
 
-    public intersect(
-        x: number, y: number, w: number = 0, h: number = 0
-    ): boolean {
-        // First, check bounding box
-        if (!super.intersect(x, y, w, h))
-            return false;
+}
 
-        // Then (if point), check distance from line
-        if (w == 0 || h == 0) {
-            for (let i = 0; i < this.points.length - 1; i++) {
-                const dist = ptLineDistance(
-                    { x: x, y: y }, this.points[i], this.points[i + 1]
+export class InterstateEdge extends Edge {
+
+    public create_arrow_line(ctx: CanvasRenderingContext2D): void {
+        ctx.moveTo(this.points[0].x, this.points[0].y);
+        let lastX = this.points[0].x;
+        let lastY = this.points[0].y;
+        let i;
+        for (i = 1; i < this.points.length; i++) {
+            const intermediateY = (lastY + this.points[i].y) / 2.0;
+            ctx.bezierCurveTo(
+                lastX, intermediateY,
+                this.points[i].x, intermediateY,
+                this.points[i].x, this.points[i].y
+            );
+            lastX = this.points[i].x;
+            lastY = this.points[i].y;
+        }
+    }
+
+    protected drawArrow(
```

Comment:
It accomodates the two different styles of interpolating the arrow points for interstate edges and memlets. Memlets use the traditional curve style, which requires the arrow heads to be angled depending on the angle with which the arrow line intersects with the destination element. Interstate edges now use a slightly different interpolation style that makes arrows less 'curvy' and more straight (shown in the image above). Since this changes where the arrow points are in relation to the actual line of the arrow, this requires a different way of calculating the angle with which the arrow head needs to be angled.

Reply:
perhaps it is good to add a comment to the code then